tests/integration/test_application.py

```diff
@@ -147,26 +137,24 @@ async def test_deploy_charmhub_charm(event_loop):
 @pytest.mark.asyncio
 async def test_upgrade_charm(event_loop):
     async with base.CleanModel() as model:
-        app = await model.deploy('cs:ubuntu-0')
+        app = await model.deploy('ubuntu')
         await model.block_until(lambda: (len(app.units) > 0 and
                                          app.units[0].machine))
-        assert app.data['charm-url'] == 'cs:ubuntu-0'
+        assert app.data['charm-url'] == 'ubuntu'
```

Comment:
this should probably be something like `assert 'ubuntu' in app.data['charm-url']` as the url will be something like `ch:amd64/focal/ubuntu-21`, but more importantly, after removing the `cs:` stuff this test is a tautology unless we try to deploy an earlier version of the charm. It currently tests only that the `app.upgrade_charm()` call doesn't error.

We might be able to use the `juju-qa-test` charm for this, e.g. deploy from the `2.0/stable` channel and upgrade to `2.0/edge`, that'll bring it from rev 22 to rev 23 (`juju info juju-qa-test`).

Reply:
I have removed this in favour of `test_upgrade_charm_switch_channel`.